core/message-classification/src/main/java/uk/gov/dwp/queue/triage/core/classification/action/ResendFailedMessageAction.java

```diff
@@ -1,13 +1,40 @@
 package uk.gov.dwp.queue.triage.core.classification.action;
 
+import com.fasterxml.jackson.annotation.JacksonInject;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import uk.gov.dwp.queue.triage.core.domain.FailedMessage;
+import uk.gov.dwp.queue.triage.core.domain.update.StatusUpdateRequest;
+import uk.gov.dwp.queue.triage.core.service.FailedMessageService;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.util.Optional;
+
+import static uk.gov.dwp.queue.triage.core.domain.StatusHistoryEvent.Status.RESEND;
 
 public class ResendFailedMessageAction implements FailedMessageAction {
 
+    private final Duration resendDelay;
+    private final FailedMessageService failedMessageService;
 
+    public ResendFailedMessageAction(@JsonProperty("resendDelay") Duration resendDelay,
+                                     @JacksonInject FailedMessageService failedMessageService) {
+        this.resendDelay = resendDelay;
+        this.failedMessageService = failedMessageService;
+    }
 
     @Override
     public void accept(FailedMessage failedMessage) {
+        failedMessageService.update(
+                failedMessage.getFailedMessageId(),
+                new StatusUpdateRequest(RESEND, getClock().instant().plus(getResendDelay())));
+    }
+
+    private Duration getResendDelay() {
+        return Optional.ofNullable(resendDelay).orElse(Duration.ofSeconds(0));
+    }
 
+    protected Clock getClock() {
```

Comment:
@dwp-kevinpotgieter I couldn't decide between adding the `Clock` as a parameter in the constructor or the current implementation of exposing a method that can be overridden.  If I added the `Clock` into the constructor I would need to ensure that it was added into the Jackson `InjectableValues.Std` so that Jackson is able to automagically inject it.

Reply:
I'm personally a fan of passing it in as a constructor arg, that way you know for sure up-front what you need in order to work with the object in question. Adding it in so that Jackson does the automagic injection is a simple enough.